ddp-workspace/projects/ddp-dsm-ui/src/app/scanner/scanner.component.ts

```diff
@@ -67,24 +69,24 @@ export class ScannerComponent implements DoCheck, OnDestroy {
 
   public removeFields(index: number): void {
     this.scannerFields.length - 1 && this.scannerFields.removeAt(index);
+    this.shortIds.length - 1 && this.shortIds.splice(index, 1);
```

Comment:
I think this part -> `this.shortIds.length - 1` is not necessary, as according to this function:
```
 responseData.forEach((data: any, index: number) =>
      {
      data?.hasOwnProperty('shortId') && this.setShortId(index, data);
      }
    );
```
It will store the values to the `shortIds` array that was returned from the backend, which in theory, must be in the same order as the front end send those, therefore there is no need to check for the empty pair (Last ones) as they just won't be there.

Especially, if you removed this part:
`this.shortIds = Array(...Array(this.scannerFields.length)).map(function() {});`
Which I think is also not needed.

What do you think?